src/main/java/uk/gov/hmcts/dts/fact/services/admin/AdminCourtPostcodeService.java

```diff
@@ -26,11 +27,15 @@
 public class AdminCourtPostcodeService {
     private final CourtRepository courtRepository;
     private final CourtPostcodeRepository courtPostcodeRepository;
+    private final AdminAuditService adminAuditService;
 
     @Autowired
-    public AdminCourtPostcodeService(final CourtRepository courtRepository, final CourtPostcodeRepository courtPostcodeRepository) {
+    public AdminCourtPostcodeService(final CourtRepository courtRepository,
+                                     final CourtPostcodeRepository courtPostcodeRepository,
+                                     final AdminAuditService adminAuditService) {
         this.courtRepository = courtRepository;
         this.courtPostcodeRepository = courtPostcodeRepository;
+        this.adminAuditService = adminAuditService;
```

Comment:
After seeing the audit service being repeated multiple times in service constructors, I'm wondering if we should have a common base class that has this injected (called something like AuditableService) or create our own annotation and use the annotation on these classes?

Reply:
Yeah I think if its declared as an abstract class and extended, we will still need a super(the dependents that need to be autowired) in each service, so it won't be much better